release-notes-1.2.0.md

```diff
@@ -0,0 +1,761 @@
+
+```ballerina
+xml x = xml `<root>
+<person><name><fname>Kamal</fname><lname>Gunae</lname></name></person>
+<person><name><fname>Nimal</fname><lname>Jayee</lname></name></person>
+<person><name><fname>Sunil</fname><lname>Kumarae</lname></name></person>
+</root>`;
+
+// Select all children items of `x`.
+xml allChildren = x/*;
+
+// Select all children items, which are XML elements.
+xml allElementChildren = x/<*>;
+
+// Match first-level children with element name `person`.
+// Then, match the second level with element name `name`
+// and the third level with element name `fname`.
+xml fnameItems = x/<person>/<name>/<fname>;
+
+// Select all descendants, which match element name `fname`.
+xml fnameDescendents = x/**/<fname>;
+```
+
+XML filter expression allows filtering an XML sequence by an element name.
+
+```ballerina
+xml x = xml `<root>
+<rectangle length="5" width="10"/>
+<rectangle length="5" width="5"/>
+<circle radius="2"/>
+<square side="6"/>
+</root>`;
+
+xml rectangles = x/*.<rectangle>;
+```
+
+#### XML attribute access
+
+ ```ballerina
+xmlns "www.ballerina.io/ns" as ns;
+xml val = xml `<element type="fixed" ns:count="2></element>`;
+string|error 'type = val.'type;
+string|error count = val.ns:count;
+```
+
+XML attribute access is now lax typed. This means that compile-time type checking is relaxed and moved to runtime. Accessing a non-existent attribute or using field access expression on a non-XML element item will result in an error being returned. If optional field access syntax is used, then nil will be returned instead of an error when the field is not available. Attributes with namespace prefix are accessed using the `value.prefix:attrName` syntax where the XML prefix must be declared in the scope.
+
+```ballerina
+xml val = xml `<element type="fixed"></element>`;
+string|error nonEx = val.nonExsistent; // result in error
+string|error? nonExOptional = val?.nonExsistent; // result in nil
+```
+
+### New language-integrated query design
+
+#### Query expression
+
+A query expression provides a language-integrated query feature using SQL-like syntax. 
+
+In its most basic form, a query expression consists of four kinds of clauses: from, let, where, and select. The first clause must be a from clause and the last clause must be a select. The result of the query expression is a list. In this basic form, a query expression is just a list comprehension.
+
+The from clause works similarly to a foreach statement. It creates an iterator from an iterable value and then binds variables to each value returned by the iterator. The where clause is a boolean expression, which can refer to variables bound by the from clause; when the where expression evaluates to false, the iteration skips following clauses. The let clause binds variables. The select clause is evaluated for each iteration; the result of the query expression is a list whose members are the result of the select clause. 
```

Comment:
Here also the clauses need to be formatted as code since those are keywords.

Reply:
@pubudu91 - Can you edit here itself that I can commit directly?

Reply:
```suggestion
The `from` clause works similarly to a foreach statement. It creates an iterator from an iterable value and then binds variables to each value returned by the iterator. The `where` clause is a `boolean` expression, which can refer to variables bound by the `from` clause; when the `where` expression evaluates to `false`, the iteration skips following clauses. The `let` clause binds variables. The `select` clause is evaluated for each iteration; the result of the query expression is a list whose members are the result of the `select` clause. 
```